### PR TITLE
- Use a Supplier<Identifable> to construct objects from id.

### PR DESCRIPTION
**Pull request by baldersheim**
- Provide lambda for default constructing frequently used objects to avoid reflection.

@bjorncs PR

### CODE REVIEW

**Comment by baldersheim** [comment]
or @bratseth

**Review by bratseth — APPROVED** [review]
💯

File: vespajlib/src/main/java/com/yahoo/vespa/objects/Identifiable.java [hunk 1]
```diff
@@ -186,6 +185,14 @@ protected static int registerClass(int id, Class<? extends Identifiable> spec) {
         registry.add(id, spec);
         return id;
     }
```
Comment (bjorncs) [inline vespajlib/src/main/java/com/yahoo/vespa/objects/Identifiable.java:187]: Consider deprecating

**Review by bjorncs — COMMENTED** [review]
👍

**Pull request merged**